chain/chunks/src/logic.rs

```diff
@@ -46,6 +48,31 @@ pub fn cares_about_shard_this_or_next_epoch(
         || shard_tracker.will_care_about_shard(account_id, parent_hash, shard_id, is_me)
 }
 
+pub fn get_shards_cares_about_this_or_next_epoch(
+    account_id: Option<&AccountId>,
+    is_me: bool,
+    block_header: &BlockHeader,
+    shard_tracker: &ShardTracker,
+    epoch_manager: &dyn EpochManagerAdapter,
+) -> Vec<ShardId> {
```

Comment:
Let's please try to return shard_uids here instead of shard_ids. With this, we can get rid of the call to `shard_ids_to_uids` as well.

Reply:
We have the epoch manager to get the shard layout if needed.

Reply:
The place where I call `get_shards_cares_about_this_or_next_epoch` uses `Vec<ShardId>`.
But now I see I only use `get_shards_cares_about_this_or_next_epoch` once, so I will remove this function for simplicity of this PR.